main.go

```
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+)
+
+// Converts a JSON file containing people and ages to a human-readable txt file
+func main() {
+
+	type Person struct {
+		FirstName string
+		Age       int
+	}
+
+	var people []Person
+
+	data, err := ioutil.ReadFile("./testdata/people.json")
+	if err != nil {
+		fmt.Println("error:", err)
```

If there's an error reading the file, we print it out, but carry on. Instead, we should probably return this error to the caller. If this `main` function becomes something like `ReadPeople`, then we could have that function return a slice of Person and an error value, couldn't we?

---

not sure I follow

---

Well, right now we 'handle' the error by just printing it out, but we still carry on, which is wrong. There was an error reading the data, so we either have no data, incomplete data, or corrupt data. It doesn't make sense to continue in this case.

So what we should do? For now, let's use `log.Fatal` to signal a fatal error and exit.